examples/thaumcraft.groovy

```diff
@@ -0,0 +1,110 @@
+    .register()
+
+mods.thaumcraft.ArcaneWorkbench.removeByOutput(item('thaumcraft:mechanism_simple'))
+
+mods.thaumcraft.ArcaneWorkbench.recipeBuilder()
+    .input(item('minecraft:melon'), 9)
+    .output(item('minecraft:melon_block'))
+    .researchKey("UNLOCKALCHEMY@3")
+    .aspects((new AspectList()).add(Aspect.EARTH, 1))
+    .vis(5)
+    .shapeless()
+    .register()
+
+mods.thaumcraft.ArcaneWorkbench.recipeBuilder()
+        .input(item('minecraft:melon'), 1)
+        .output(item('thaumcraft:void_hoe'))
+        .researchKey("UNLOCKALCHEMY@3")
+        .aspects(new AspectList())
+        .vis(0)
+        .shapeless()
+        .register()
+
+mods.thaumcraft.ArcaneWorkbench.recipeBuilder()
+    .recipe(new Object[]{"SS ", "   ", "   ", (char)'S', item('minecraft:pumpkin_seeds')})
+    .output(item('minecraft:pumpkin'))
+    .researchKey("UNLOCKALCHEMY@3")
+    .aspects((new AspectList()).add(Aspect.EARTH, 1))
+    .vis(5)
+    .register()
+
+//new AspectWrapper.AspectBuilder()
+//        .tag("humor")
+//        .chatColor(14013676)
+//        .components(new Aspect[]{Aspect.ENTROPY, Aspect.MIND})
+//        .image(new ResourceLocation("thaumcraft", "textures/aspects/humor.png"))
+//        .register()
+
+new AspectWrapper.AspectHelper()
+    .object(item('minecraft:pumpkin'))
+    .stripAspects()
+    .aspects((new AspectList()).add(Aspect.METAL, 20))
+    .register()
+
+new Warp().addWarp(item('minecraft:pumpkin'), 3)
+new Warp().removeWarp(item('thaumcraft:void_hoe'))
+
+new DustTrigger().removeByOutput(item('thaumcraft:arcane_workbench'))
+
+new DustTrigger.TriggerBuilder()
+    .target(Blocks.OBSIDIAN)
+    .output(item('minecraft:enchanting_table'))
+    .researchKey("UNLOCKALCHEMY@3")
+    .register()
+
+new LootBag().removeAll(new int[]{0,1,2})
+new LootBag().addItem(item('minecraft:diamond_block'), 100, new int[]{0,1,2})
+
+new SmeltingBonus.SmeltingBonusBuilder()
+    .input(item('minecraft:cobblestone'))
+    .output(item('minecraft:stone_button'))
+    .chance(0.2F)
+    .register()
+
+SmeltingBonus.removeByOutput(item('minecraft:gold_nugget'))
+
+//new Research.ResearchCategoryBuilder()
+//    .key("BASICS2")
+//    .researchKey("UNLOCKAUROMANCY")
+//    .formula((new AspectList()).add(Aspect.PLANT, 5).add(Aspect.ORDER, 5).add(Aspect.ENTROPY, 5).add(Aspect.AIR, 5).add(Aspect.FIRE, 5).add(Aspect.EARTH, 3).add(Aspect.WATER, 5))
+//    .icon(new ResourceLocation("thaumcraft", "textures/aspects/humor.png"))
+//    .background(new ResourceLocation("thaumcraft", "textures/gui/gui_research_back_1.jpg"))
+//    .background2(new ResourceLocation("thaumcraft", "textures/gui/gui_research_back_over.png"))
+//    .register()
+//
+//Research.addResearchLocation(new ResourceLocation("thaumcraft", "research/new.json"))
+
+Research.addScannable("KNOWLEDGETYPEHUMOR", item('minecraft:pumpkin'))
```

Comment:
Inconsistent syntax
```suggestion
mods.thaumcraft.Research.addScannable("KNOWLEDGETYPEHUMOR", item('minecraft:pumpkin'))
```

Reply:
consistent syntax added in commit: [https://github.com/CleanroomMC/GroovyScript/commit/21691d5220732e4421f2bdf7c4083c5bf7d6cf81](https://github.com/CleanroomMC/GroovyScript/commit/21691d5220732e4421f2bdf7c4083c5bf7d6cf81)